### ISSUE TITLE
Feature Request: Optionally don't reload custom scripts

### ISSUE THREAD

**Issue by Mobius5150** [Firebottle/Firebot#337]
Howdy,

I was working on a custom script recently that uses an SDK that requires a connection be maintained. Because of that I got wonky behavior every time the button was pressed in Firebot as the script was unloaded. It would be really great if a script could specify that it didn't want to be reloaded every time. 

E.g.
```javascript
exports.reloadOnButtonPressed = false;
```
or
```javascript
exports.reloadOnButtonPressed = function() {
    return new Promise((resolve, reject) => {
        resolve(false);
    });
};
```

If you like the feature idea and let me know what you think the method signature should be I'd be happy to submit a PR.

**Comment by Firebottle** [comment]
We're happy to welcome all mixer community devs. Feel free to submit a PR and as long as it doesnt break current functionality everything should be good. :)

**Comment by ebiggz** [comment]
I've implemented this complete. Decided to just have a setting in the "Advanced" tab that lets the user turn off or on the clearing of the script cache. This setting only shows if you have the "enable custom script" setting enabled as well. It defaults to off.

**Comment by Firebottle** [comment]
Released in the dank update 4.2.0
